### DISCUSSION TITLE
Help adding comment to an image

### DISCUSSION THREAD

**Discussion by Tubby76** [dlemstra/Magick.NET#1408] [Help]
Hi

Could someone help me with adding a comment to an image please.

I've successfully read an existing comment that I manually added to a JPG but I can't seem to set it programmatically.

Please see my code snippet below - the image is saved but the comment is not written.

```
Using ImageToProcess As New MagickImage(SourceFile.File)
            ImageToProcess.Comment = "test"
            ImageToProcess.Write(_DestinationDir.ToString & SourceFile.File.Name)

End Using
```

Thanks

**Comment by dlemstra** [comment]
This is the correct code to add a comment to the image. Are you sure that your output image is a `jpeg` file? Can you share your input image?

**Comment by Tubby76** [comment]
Sure, please find input/output attached - the output has been resized to a larger size and saved using Magick. The original comment remains in the file.

I'm using Magick.NET.Core and Magick.NET-Q8-x64, both versions are v13.1.3

Thanks

![test-input](https://github.com/dlemstra/Magick.NET/assets/138673653/469431d2-4628-476c-9fbd-a873d72b055a)
![test-output](https://github.com/dlemstra/Magick.NET/assets/138673653/92330a95-0571-411c-8be1-038adcf17e9b)

**Comment by Tubby76** [comment]
Is anybody able to verify that adding a comment works please?

**Comment by dlemstra** [comment]
Your file contains a comment with the value "test".

**Comment by Tubby76** [comment]
After changing the comments I was expecting the comments field in Windows file explorer to have updated, but as can be seen in this image it's showing the old comment:

![image](https://github.com/dlemstra/Magick.NET/assets/138673653/ea58fbb2-8969-488b-a2f1-8146d1e6f221)

I guess Magick comments are stored elsewhere in the file?

**Reply by dlemstra** [reply]
Those comments might be stored somewhere else. Maybe the exif profile?

**Reply by dlemstra** [reply]
Just did a check and this is indeed stored in the exif profile.
